tests/nlu/featurizers/test_lexical_syntactic_featurizer.py

```diff
@@ -1,197 +1,288 @@
 import numpy as np
 import pytest
-
-import scipy.sparse
-from typing import Text
-
-from rasa.nlu.tokenizers.spacy_tokenizer import SpacyTokenizer
-from rasa.nlu.tokenizers.whitespace_tokenizer import WhitespaceTokenizer
+import re
+from typing import Text, Dict, Any, Callable, List, Optional
+
+from rasa.engine.graph import ExecutionContext
+from rasa.engine.storage.storage import ModelStorage
+from rasa.engine.storage.resource import Resource
+from rasa.nlu.constants import (
+    DENSE_FEATURIZABLE_ATTRIBUTES,
+    MESSAGE_ATTRIBUTES,
+    TOKENS_NAMES,
+)
+from rasa.nlu.tokenizers.spacy_tokenizer import POS_TAG_KEY
 from rasa.nlu.featurizers.sparse_featurizer.lexical_syntactic_featurizer import (
-    LexicalSyntacticFeaturizer,
+    LexicalSyntacticFeaturizerGraphComponent,
+    FEATURES,
 )
 from rasa.shared.nlu.training_data.training_data import TrainingData
 from rasa.shared.nlu.training_data.message import Message
-from rasa.nlu.constants import SPACY_DOCS
-from rasa.shared.nlu.constants import TEXT, ACTION_TEXT
+from rasa.shared.nlu.constants import FEATURE_TYPE_SEQUENCE, TEXT
+from rasa.shared.exceptions import InvalidConfigException
+from rasa.nlu.tokenizers.tokenizer import Token
+
+
+@pytest.fixture
+def resource_lexical_syntactic_featurizer() -> Resource:
+    return Resource("LexicalSyntacticFeaturizerGraphComponent")
+
+
+@pytest.fixture
+def create_lexical_syntactic_featurizer(
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    resource_lexical_syntactic_featurizer: Resource,
+) -> Callable[[Dict[Text, Any]], LexicalSyntacticFeaturizerGraphComponent]:
+    def inner(config: Dict[Text, Any]):
+        return LexicalSyntacticFeaturizerGraphComponent.create(
+            config=config,
+            model_storage=default_model_storage,
+            execution_context=default_execution_context,
+            resource=resource_lexical_syntactic_featurizer,
+        )
+
+    return inner
 
 
 @pytest.mark.parametrize(
-    "sentence, expected_features",
+    "sentence,part_of_speech,feature_config,expected_features",
     [
+        # simple example 1
         (
             "hello goodbye hello",
+            None,
+            [["BOS", "upper"], ["BOS", "EOS", "prefix2", "digit"], ["EOS", "low"],],
             [
                 [0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0],
                 [0.0, 1.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0],
                 [1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0],
-                [1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 1.0, 3.0, 1.0, 2.0, 1.0, 1.0, 2.0],
```

Comment:
These were CLS token features that are no longer computed?

Reply:
they were never computed - the old comparison had a "[:-1]" to exclude these and checked whether sentence embeddings are None

Reply:
... so actually yeah, probably at some point there was some "sum" aggregation it seems :D

Reply:
> so actually yeah, probably at some point there was some "sum" aggregation it seems

Sorry, I'm confused. So, we should add the "sum" aggregation or we shouldn't? :)

Reply:
Okay I see what you mean.. The test originally just excluded the last row of the expected array anyways 🤦

Reply:
yep, I was surprised as well :)